views.py

```diff
@@ -204,11 +204,23 @@ def parse_MetabolomicsWorkbench(usi):
 
 
 def _prepare_spectrum(usi, **kwargs):
-    masses, intensities = zip(*parse_USI(usi)['peaks'])
+    usi_data = parse_USI(usi)
+    masses, intensities = zip(*usi_data['peaks'])
+    precursor_mz = usi_data.get('precursor_mz',0)
+    if kwargs.get('plot_precursor',False) and precursor_mz > 0:
```

Comment:
Maybe instead of adding the precursor as an actual peak it'd be better to explicitly draw it after the spectrum has been plot? That way we have more control about how it's plot and we can differentiate it from the other peaks visually. It's only a virtual peak after all, and I think we should make that explicitly clear through coloring/dotted lines/whatever.

Also, if there's a precursor loss peak then adding the precursor peak again will probably be somewhat messy, so we should account for that.

Reply:
Yep, fair enough...adding it after makes sense (if we want it at all)

Reply:
maybe we should ditch this request?